spec/ssh_spec.rb

```diff
@@ -0,0 +1,106 @@
+# vim: syntax=ruby:expandtab:shiftwidth=2:softtabstop=2:tabstop=2
+
+# Copyright 2020-present Facebook
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'logger'
+require 'base64'
+require 'between_meals/util'
+require 'taste_tester/logging'
+require 'taste_tester/exceptions'
+require 'taste_tester/ssh_util'
+require 'taste_tester/ssh'
+require 'taste_tester/config'
+
+describe TasteTester::SSH do
+  let(:logger) do
+    Logger.new('/dev/null')
+  end
+
+  tt_ssh = TasteTester::SSH.new('mock_host')
+
+  context 'test default config' do
+    it 'test ssh base command' do
+      expect(tt_ssh.ssh_base_cmd).to eq(
+        'ssh  ' \
+        '-T ' \
+        '-o BatchMode=yes ' \
+        '-o UserKnownHostsFile=/dev/null ' \
+        '-o StrictHostKeyChecking=no ' \
+        '-o ConnectTimeout=5',
+      )
+    end
+
+    it 'test ssh target' do
+      expect(tt_ssh.ssh_target).to eq(
+        'root@mock_host',
+      )
+    end
+
+    it 'test build ssh command' do
+      expect(tt_ssh.build_ssh_cmd('mock_ssh', ['cmd1', 'cmd2'])).to eq(
+        'mock_ssh root@mock_host ' \
+        "\"echo 'Y21kMSAmJiBjbWQy' | base64 --decode | bash -x\"",
+      )
+    end
+  end
+
+  context 'test custom configs linux' do
+    before(:each) do
```

Comment:
why before _each_ ? Unless you're changing these in tests, you can just do `before do`